storage/Filesystems/BtrfsImpl.cc

```diff
@@ -78,6 +78,32 @@ namespace storage
     }
 
 
+    BtrfsSubvolume*
+    Btrfs::Impl::get_default_btrfs_subvolume()
+    {
+	for (BtrfsSubvolume* btrfs_subvolume : get_btrfs_subvolumes())
+	{
+	    if (btrfs_subvolume->is_default_btrfs_subvolume())
+		return btrfs_subvolume;
+	}
+
+	ST_THROW(Exception("no default btrfs subvolume found"));
+    }
+
+
+    const BtrfsSubvolume*
+    Btrfs::Impl::get_default_btrfs_subvolume() const
+    {
+	for (const BtrfsSubvolume* btrfs_subvolume : get_btrfs_subvolumes())
+	{
+	    if (btrfs_subvolume->is_default_btrfs_subvolume())
+		return btrfs_subvolume;
+	}
+
+	ST_THROW(Exception("no default btrfs subvolume found"));
+    }
+
```

Comment:
Is it possible to avoid to write exactly the same implementation twice? Maybe using `const_cast` ?